components/locid_transform/src/canonicalizer.rs

```diff
@@ -260,39 +254,74 @@ impl LocaleCanonicalizer {
     pub const fn new_with_expander(expander: LocaleExpander) -> Self {
         Self {
             aliases: DataPayload::from_static_ref(
-                crate::provider::Baked::SINGLETON_LOCID_TRANSFORM_ALIASES_V1,
+                crate::provider::Baked::SINGLETON_LOCID_TRANSFORM_ALIASES_V2,
             ),
             expander,
         }
     }
 
+    fn try_new_with_expander_compat<P>(
+        provider: &P,
+        expander: LocaleExpander,
+    ) -> Result<Self, LocaleTransformError>
+    where
+        P: DataProvider<AliasesV2Marker> + DataProvider<AliasesV1Marker> + ?Sized,
+    {
+        let payload_v2: Result<DataPayload<AliasesV2Marker>, _> = provider
+            .load(Default::default())
+            .and_then(DataResponse::take_payload);
+        let aliases = if let Ok(payload) = payload_v2 {
+            payload
+        } else {
+            let payload_v1: DataPayload<AliasesV1Marker> = provider
+                .load(Default::default())
+                .and_then(DataResponse::take_payload)?;
+            payload_v1.try_map_project(|st, _| st.try_into())?
+        };
+
+        Ok(Self { aliases, expander })
+    }
+
     #[doc = icu_provider::gen_any_buffer_unstable_docs!(UNSTABLE, Self::new_with_expander)]
     pub fn try_new_with_expander_unstable<P>(
         provider: &P,
         expander: LocaleExpander,
-    ) -> Result<LocaleCanonicalizer, LocaleTransformError>
+    ) -> Result<Self, LocaleTransformError>
     where
-        P: DataProvider<AliasesV1Marker> + ?Sized,
+        P: DataProvider<AliasesV2Marker> + ?Sized,
     {
-        let aliases: DataPayload<AliasesV1Marker> =
+        let aliases: DataPayload<AliasesV2Marker> =
             provider.load(Default::default())?.take_payload()?;
 
-        Ok(LocaleCanonicalizer { aliases, expander })
+        Ok(Self { aliases, expander })
     }
 
-    icu_provider::gen_any_buffer_data_constructors!(
-        locale: skip,
+    #[cfg(feature = "compiled_data")]
+    #[cfg(skip)]
+    #[doc = ""]
+    #[doc = " ✨ *Enabled with the `compiled_data` Cargo feature.*"]
+    #[doc = ""]
+    #[doc = " [📚 Help choosing a constructor](icu_provider::constructors)"]
+    pub fn new_with_expander(options: LocaleExpander) -> Result<Self, LocaleTransformError> {
+        Self::try_new_with_expander_unstable(&crate::provider::Baked, options)
+    }
+
+    #[doc = icu_provider::gen_any_buffer_unstable_docs!(ANY, Self::try_new_with_expander_with_buffer_provider)]
+    pub fn try_new_with_expander_with_any_provider(
```

Comment:
Nit: I think the arguments to this macro might be wrong. Please double-check. Also below.